api/restHandler/app/pipeline/configure/DeploymentPipelineRestHandler.go

```diff
@@ -531,49 +531,79 @@ func (handler *PipelineConfigRestHandlerImpl) ChangeChartRef(w http.ResponseWrit
 		common.WriteJsonResp(w, err, request, http.StatusBadRequest)
 		return
 	}
+	var envMetrics bool
+	isEnvironmentOverriden := true
 	envConfigProperties, err := handler.propertiesConfigService.GetLatestEnvironmentProperties(request.AppId, request.EnvId)
-	if err != nil || envConfigProperties == nil {
-		handler.Logger.Errorw("env properties not found, ChangeChartRef", "err", err, "payload", request)
-		common.WriteJsonResp(w, err, "env properties not found", http.StatusNotFound)
-		return
-	}
-	if !envConfigProperties.IsOverride {
-		handler.Logger.Errorw("isOverride is not true, ChangeChartRef", "err", err, "payload", request)
-		common.WriteJsonResp(w, err, "specific environment is not overriden", http.StatusUnprocessableEntity)
-		return
-	}
-	compatible, oldChartType, newChartType := handler.chartRefService.ChartRefIdsCompatible(envConfigProperties.ChartRefId, request.TargetChartRefId)
-	if !compatible {
-		common.WriteJsonResp(w, fmt.Errorf("charts not compatible"), "chart not compatible", http.StatusUnprocessableEntity)
-		return
-	}
-
-	envConfigProperties.EnvOverrideValues, err = handler.chartService.PatchEnvOverrides(envConfigProperties.EnvOverrideValues, oldChartType, newChartType)
 	if err != nil {
-		common.WriteJsonResp(w, err, "error in patching env override", http.StatusInternalServerError)
+		handler.Logger.Errorw("error in finding env properties ChangeChartRef", "err", err, "payload", request)
+		common.WriteJsonResp(w, err, "env properties not found", http.StatusInternalServerError)
 		return
 	}
+	if envConfigProperties == nil {
```

Comment:
use if else here

Reply:
In this case, If the environment is already overridden and  the users sets the value of the flag as true, then it will assign the default values to the deployment template. Is this expected? If so, then we need to inform the user beforehand otherwise their overridden values will get lost. One way to do it is to make the name of the flag more descreptive. Just need your confirmation on the same @nishant-d